src/simplex.h

```diff
@@ -3,8 +3,79 @@
 
 #include <optional>
 #include "matrix.h"
+#include "algorithm"
 #include "vector_ops.h"
 
+#define MALFORMED_INPUT 0
+#define DEGENERACY 1
+#define ALTERNATIVE_OPTIMA 2
+#define UNBOUNDED 3
+
+class ValidationReport {
+private:
+    bool OK;
+    // 1 - Degeneracy; 2 - Alternative Optima; 3 - Unbounded Solution;
+    // 0 - Incorrect Input (Mismatch number of base vars.)
+    int specialCaseNum;
+    std::vector<size_t> incorrectVars;
+
+    friend std::ostream& operator<<(std::ostream&, const ValidationReport&);
+public:
+    ValidationReport(bool isOK,
+                     int specialCase,
+                     const std::vector<size_t>& incorrectVars) :
+                     OK(isOK), specialCaseNum(specialCase), incorrectVars(incorrectVars) {}
+
+    [[nodiscard]] bool is_OK() const {
+        return OK;
+    }
+
+    [[nodiscard]] int specialCase() const {
+        return specialCaseNum;
+    }
+
+    [[ nodiscard ]] std::vector<size_t> get_incorrectVars() const {
+        return incorrectVars;
+    }
+};
+
+std::ostream& operator<<(std::ostream& stream, const ValidationReport& report) {
+    if (report.is_OK()) {
+        return (stream << "");
+    }
+
+    if (report.specialCaseNum == MALFORMED_INPUT) {
+        stream << ("There are incorrect base variables. Recheck the columns of variables.\n");
+
+        stream << "Found basic variables: ";
+        for (unsigned long long incorrectVar : report.incorrectVars) {
+            stream << "X_" << incorrectVar << "; ";
+        }
+        stream << std::endl;
+
+        return stream;
+    }
+
+    stream << ("[ SPECIAL CASE ]\n");
+
+    switch (report.specialCase()) {
+        case DEGENERACY: stream << ("There is a tie for a minimum ratio, which can increase the infinite loop.\n");
```

Comment:
Looks like you may need to do `break;` in `switch` branches. In C++ you fall through all the bottom cases if got into one above. May be you need to prevent it here